src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +705,227 @@ public String[] executeSimple(Object[][] params) {
 						"\t\tset {_money} to formatNumber({money::%sender's uuid%})",
 						"\t\tsend \"Your balance: %{_money}%\" to sender")
 			.since("2.10");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("mean", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				double total = 0;
+				for (Object object : params[0]) {
+					Number number = (Number) object;
+					if (number instanceof Double doubleNumber && (doubleNumber.isInfinite() || doubleNumber.isNaN()))
+						return null;
+					total += number.doubleValue();
+				}
+				total /= params[0].length;
```

Comment:
Also need to ensure that total itself doesn't become infinite. (could technically allow bigger sums if you divide prior to adding each number, but come on, who's getting the means of 1.7e308)